unit_tests/test_tlslite_recordlayer.py

```diff
@@ -514,6 +515,209 @@ def test_sendRecord_with_encrypting_set_up_tls1_2(self):
+            else:
+                break
+        self.assertEqual(len(sock.sent), 1)
+        self.assertEqual(sock.sent[0][:5], bytearray(
+            b'\x17' +  # application_data
+            b'\x03\x01' +  # hidden protocol version - TLS 1.x
+            b'\x00\x15'  # length
+            ))
+        self.assertEqual(sock.sent[0][5:], bytearray(
+            b'}\x17w_#\xf0\xf2R\xaa*s\xe2\xca\xab\x9d\xea\x9d\xf3\xc1-\xd2'
+            ))
+
+    def test_sendRecord_with_encryption_tls1_3_chacha20(self):
+        patcher = mock.patch.object(os,
+                                    'urandom',
+                                    lambda x : bytearray(x))
+        mock_random = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        sock = MockSocket(bytearray(0))
+
+        recordLayer = RecordLayer(sock)
+        recordLayer.version = (3, 4)
+
+        ciph = CipherSuite.TLS_CHACHA20_POLY1305_SHA256
+        recordLayer.calcTLS1_3PendingState(ciph,
+                                           bytearray(48),  # cl_traffic_sec
+                                           bytearray(48),  # sr_traffic_sec
+                                           None)  # implementations
+        recordLayer.changeWriteState()
+
+        app_data = ApplicationData().create(b'test')
+
+        for result in recordLayer.sendRecord(app_data):
+            if result in (0, 1):
+                self.assertTrue(False, "blocking socket")
+            else:
+                break
+        self.assertEqual(len(sock.sent), 1)
+        self.assertEqual(sock.sent[0][:5], bytearray(
+            b'\x17' +  # application_data
+            b'\x03\x01' +  # hidden protocol version - TLS 1.x
+            b'\x00\x15'  # length
+            ))
+        self.assertEqual(sock.sent[0][5:], bytearray(
+            b'o\x9fO\x16\x07\x878]GV\xa5l\x12\xb6\x85\xb5@\x83\x94\x06\xd6'
+            ))
+
+    def test_sendRecord_with_malformed_inner_plaintext(self):
+        # setup
+        patcher = mock.patch.object(os,
+                                    'urandom',
+                                    lambda x : bytearray(x))
+        mock_random = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        sock = MockSocket(bytearray(0))
+
+        recordLayer = RecordLayer(sock)
+        recordLayer.version = (3, 4)
+        self.assertEqual((3, 4), recordLayer.version)
+
+        recordLayer.calcTLS1_3PendingState(CipherSuite.TLS_AES_128_GCM_SHA256,
+                                           bytearray(32),  # cl_traffic_sec
+                                           bytearray(32),  # sr_traffic_sec
+                                           None)  # implementations
+        recordLayer.changeWriteState()
+
+        app_data = ApplicationData().create(b'\x00\x00\x00\x00')
+        app_data.contentType = 0
+
+        for result in recordLayer.sendRecord(app_data):
+            if result in (0, 1):
+                self.assertTrue(False, "blocking socket")
```

Comment:
The same as in previous comment: `assertTrue(False)` sounds confusing (at least without any comment). The same is once more few lines below.